src/components/BorrowerProfile/BorrowerSocialShare.vue

```diff
@@ -0,0 +1,293 @@
+				class: '',
+				disabled: false,
+				text: ''
+			},
+			shareMessage: '',
+			shareLink: `https://${this.$appConfig.host}${this.$route.path}`
+		};
+	},
+	computed: {
+		facebookShareUrl() {
+			return this.getFullUrl('https://www.facebook.com/dialog/share', {
+				app_id: this.$appConfig.fbApplicationId,
+				display: 'page',
+				href: `${this.shareLink}?utm_source=facebook.com&utm_medium=social&utm_campaign=social_share_checkout`,
+				redirect_uri: `${this.shareLink}?kiva_transaction_id=${this.$route.query.kiva_transaction_id}`,
+				quote: this.shareMessage,
+			});
+		},
+		emailShareUrl() {
+			return this.getFullUrl('Mailto:', {
+				Subject: `Help fund ${this.borrowerName} 's ${this.sector} loan through Kiva!`,
+				// eslint-disable-next-line max-len
+				Body: `Hello,\n\nI thought you might be interested in supporting this Kiva loan, ${this.shareLink}.\n\n`
+					+ `Even a small amount could help fund ${this.borrowerName}'s loan. And if you can't make a `
+					+ 'donation, it would be great if you could share the fundraiser to help spread the word.'
+					+ '\n\nThanks for taking a look!.',
+			});
+		},
+		twitterShareUrl() {
+			return this.getFullUrl('https://twitter.com/intent/tweet', {
+				text: this.shareMessage,
+				url: `${this.shareLink}?utm_source=t.co&utm_medium=social&utm_campaign=social_share_checkout`,
+				via: 'Kiva',
+			});
+		},
+	},
+	methods: {
+		getFullUrl(base, args) {
+			const querystring = _map(args, (val, key) => `${key}=${encodeURIComponent(val)}`).join('&');
+			return `${base}?${querystring}`;
+		},
+		handleFacebookResponse() {
+			// Check for the route hash that facebook adds to the request
+			if (this.$route.hash === '#_=_') {
+				// Check for an error
+				const { error_code: code, error_message: message } = this.$route.query;
+				if (code) {
+					// The 4201 error code means the user pressed 'Cancel', so can be ignored
+					if (code !== '4201') {
+						this.$showTipMsg(`There was a problem sharing to Facebook: ${message}`, 'warning');
+					}
+				} else {
+					this.$showTipMsg('Thanks for sharing to Facebook!');
+				}
+			}
+		},
+		async copyLink() {
+			const url = `${this.shareLink}?utm_source=social_share_link&utm_campaign=social_share_checkout`;
+			try {
+				await clipboardCopy(url);
+				this.copyStatus = {
+					class: 'social__btn--success',
+					disabled: true,
+					text: 'Copied!'
+				};
+			} catch (err) {
+				this.copyStatus = {
+					class: 'social__btn--error',
+					disabled: true,
+					text: 'Error'
+				};
+			} finally {
+				setTimeout(() => {
+					this.copyStatus = {
+						class: '',
+						disabled: false,
+						text: ''
+					};
+				}, 2500);
+			}
+		}
+	},
+	mounted() {
+		this.handleFacebookResponse();
+	},
+};
+</script>
+
+<style lang="scss" scoped>
+@import 'settings';
+@import "foundation";
+
+$color-facebook: #3b5998;
+$color-twitter: #08a0e9;
+$color-email: #AB4147;
+$color-clipboard: #DEB13C;
+
+// layout of blocks
+.share {
```

Comment:
Our Foundation settings and Tailwind settings use different breakpoints, so I think this block of rules should be applied using Tailwind classes instead, as the rest of the page is using the Tailwind breakpoints.

Reply:
Hey @emuvente I'm trying to translate the css to tw as `tw-my-[21px] tw-px-[15px] tw-max-w-[600px] md:tw-bg-primary md:tw-p-4 md:tw-rounded`, but it seems that arbitrary values don't work if I don't set `mode: 'jit'` in tailwind.config.js.

Should I modify the config file or change the breakpoint in css to be the same as tailwind's?

Reply:
Don't mind me. It works fine without it! Cache I guess

Reply:
The max-width can be set in an inline-style on the element `style="max-width:600px;"`, and for the margin and padding I would recommend rounding to the nearest built-in classes `tw-my-2.5 tw-px-2`.

Reply:
Yes, the dev cache will get you sometimes with these.

We typically don't like to use this dynamic value approach. Rather when we have to break out of our preset size classes, we would just add them to a `style=""` attribute instead.

I don't think we should be generating classes for 21 and 15 which are really odd values against our sizing scale which is based on multiple of 4px. The design should be adjusted to use our sizing scale so you can use our preset classes.

Reply:
I see... great to know that! I'll change it right away. Thank you both!